### **Modelos de Machine Learning utilizando a abordagem BoW (Bag of Words)**

#### Importando os pacotes necessários

In [1]:
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import GaussianNB
from sklearn.naive_bayes import MultinomialNB
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import sys
sys.path.append('/home/guilherme/AIEnvironment/SentimentAnalysis/src/modeling')
sys.path.append('/home/guilherme/AIEnvironment/SentimentAnalysis/src/data_preparation')
from bag_of_words import classification_text_bow
from data_exploration import load_data

In [2]:
df = load_data('/home/guilherme/AIEnvironment/SentimentAnalysis/data/processed/df_imdb.csv')
df.head()

Reviews  Sentiment
0  Once again Mr. Costner has dragged out a movie...          0
1  This is an example of why the majority of acti...          0
2  First of all I hate those moronic rappers, who...          0
3  Not even the Beatles could write songs everyon...          0
4  Brass pictures movies is not a fitting word fo...          0

#### Modelo de Regressão Logística - Sem tratamento

In [3]:

lg_model = classification_text_bow(df, 'Reviews', 'Sentiment', LogisticRegression())
print(f"Acurácia modelo de Regressão Logística: {(lg_model*100):.2f}%")

Acurácia modelo de Regressão Logística: 73.21%


#### Modelo de Naive Bayes Gaussiano - Sem tratamento

In [4]:
nv_model = classification_text_bow(df, 'Reviews', 'Sentiment', GaussianNB())
print(f"Acurácia modelo de Naive Bayes Gaussiano: {(nv_model*100):.2f}%")

Acurácia modelo de Naive Bayes Gaussiano: 68.02%


#### Modelo de Naive Bayes Multinomial - Sem tratamento

In [8]:
nv_model = classification_text_bow(df, 'Reviews', 'Sentiment', MultinomialNB())
print(f"Acurácia modelo de Naive Bayes Multinominal: {(nv_model*100):.2f}%")

Acurácia modelo de Naive Bayes Multinominal: 68.25%


#### Modelo de Random Forest - Sem tratamento

In [7]:
rf_model = classification_text_bow(df, 'Reviews', 'Sentiment', RandomForestClassifier())
print(f"Acurácia modelo de Random Forest: {rf_model*100:.2f}%")

Acurácia modelo de Random Forest: 71.49%


#### Modelo de SVM - Sem tratamento

In [9]:
svc_model = classification_text_bow(df, 'Reviews', 'Sentiment', SVC())
print(f"Acurácia modelo de Support Vector Machine: {svc_model*100:.2f}%")

Acurácia modelo de Support Vector Machine: 73.69%


#### Avaliação dos modelos - Sem tratamento

- Utilizando apenas a métrica de acurácia para avaliação dos modelos, já que as classes encontram-se balanceadas, os modelos tiveram desempenhos baixos e semelhantes entre si, variando de 68.02% até 73.69%. Portanto, a abordagem BoW não foi eficiente para a classificação dos textos.